### PR TITLE
EI-490/EI-732 bug-fix: log learning not found reason at learner data refresh

### PR DESCRIPTION
**Pull request by ruslanss**

### CODE REVIEW

**Review by David-B-Read — APPROVED** [review]

File: src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs [hunk 1]
```diff
@@ -11,28 +10,37 @@ namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.RefreshLea
...
+            var learner = await _learnerDomainRepository.GetOrCreate(incentive);
 
-            Learner learner = await _learnerDomainRepository.GetOrCreate(incentive);
+            _logger.LogInformation("Start Learner data refresh from Learner match service for ApprenticeshipIncentiveId: {ApprenticeshipIncentiveId}, ApprenticeshipId: {ApprenticeshipId}, UKPRN: {UKPRN}, ULN: {ULN}",
+                learner.ApprenticeshipIncentiveId, learner.ApprenticeshipId, learner.Ukprn, learner.UniqueLearnerNumber);
 
             SubmissionData submissionData = null;
             var learnerData = await _learnerService.Get(learner);
+
+            _logger.LogInformation("End Learner data refresh from Learner match service for ApprenticeshipIncentiveId: {ApprenticeshipIncentiveId}, ApprenticeshipId: {ApprenticeshipId}, UKPRN: {UKPRN}, ULN: {ULN}",
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:41]: This isn't needed to support the requirement.

File: src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs [hunk 2]
```diff
@@ -47,6 +55,13 @@ public async Task Handle(RefreshLearnerCommand command, CancellationToken cancel
 
             learner.SetLearningPeriods(learnerData.LearningPeriods(incentive));
 
+
+            if (learner.SubmissionData != null && !learner.SubmissionData.LearningFoundStatus.LearningFound)
+            {
+                _logger.LogInformation("Matching ILR record not found for ApprenticeshipIncentiveId: {ApprenticeshipIncentiveId}, ApprenticeshipId: {ApprenticeshipId}, UKPRN: {UKPRN}, ULN: {ULN} with reason: {NotFoundReason}",
+                    learner.ApprenticeshipIncentiveId, learner.ApprenticeshipId, learner.Ukprn, learner.UniqueLearnerNumber, learner.SubmissionData.LearningFoundStatus.NotFoundReason);
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:62]: This is the only one required to meet the requirement.

**Review by Stephek — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs [hunk 3]
```diff
@@ -11,28 +10,37 @@ namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.RefreshLea
...
         }
 
         public async Task Handle(RefreshLearnerCommand command, CancellationToken cancellationToken = default)
         {
-            Domain.ApprenticeshipIncentives.ApprenticeshipIncentive incentive = await _incentiveDomainRepository.Find(command.ApprenticeshipIncentiveId);
+            var incentive = await _incentiveDomainRepository.Find(command.ApprenticeshipIncentiveId);
+            var learner = await _learnerDomainRepository.GetOrCreate(incentive);
 
-            Learner learner = await _learnerDomainRepository.GetOrCreate(incentive);
+            _logger.LogInformation("Start Learner data refresh from Learner match service for ApprenticeshipIncentiveId: {ApprenticeshipIncentiveId}, ApprenticeshipId: {ApprenticeshipId}, UKPRN: {UKPRN}, ULN: {ULN}",
```
[the remark this answers is not in the document]
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:35]: Additional logging added as per request from test team and personal experience while supporting and debugging issues in TEST

File: src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs (hunk 1 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:41]: Additional logging added as per request from test team and personal experience while supporting and debugging issues in TEST

File: src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs (hunk 2 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:62]: Additional logging added as per request from test team and personal experience while supporting and debugging issues in TEST

File: src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Stephek) [reply src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:35]: There is already logging as a cross cutting concern.  Add back the decorator on the service call for this type of logging.

[the remark this answers is not in the document]
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/RefreshLearner/RefreshLearnerCommandHandler.cs:35]: can't as the DTO object hasn't got all the info we want to log 🤷‍♂️
we'll be removing this additional logging after test passed

**Pull request merged**